src/layout/layout.rs

```diff
@@ -889,12 +1149,138 @@ impl Element {
     }
 }
 
+#[derive(Debug)]
+struct StrengthSet {
+    pub spacer_size_equality: f64,
+    pub proportional_scaling_equality: f64,
+    pub fixed_value_equality: f64,
+    pub min_max_inequality: f64,
+    pub length_equality: f64,
+    pub percentage_equality: f64,
+    pub ratio_equality: f64,
+    pub min_max_equality: f64,
+    pub proportional_grower: f64,
+    pub grower: f64,
+    pub space_grower: f64,
+}
+
+impl StrengthSet {
+    pub fn is_valid(&self) -> bool {
+        self.spacer_size_equality > self.fixed_value_equality
+            && self.proportional_scaling_equality > self.fixed_value_equality
+            && self.fixed_value_equality > self.min_max_inequality
+            && self.min_max_inequality > self.min_max_equality
+            && self.min_max_inequality > self.length_equality
+            && self.length_equality > self.percentage_equality
+            && self.percentage_equality > self.ratio_equality
+            && self.ratio_equality > self.min_max_equality
+            && self.min_max_inequality > self.proportional_grower
+            && self.proportional_grower > self.grower
+            && self.grower > self.space_grower
+    }
+}
+
+impl Default for StrengthSet {
+    fn default() -> Self {
+        let (
+            // ┌     ┐┌───┐┌     ┐┌───┐┌     ┐
+            //   ==x  │   │  ==x  │   │  ==x
+            // └     ┘└───┘└     ┘└───┘└     ┘
+            spacer_size_equality,
+            // ┌───────────────┐┌───────────────┐
+            // │Proportional(x)││Proportional(x)│
+            // └───────────────┘└───────────────┘
+            proportional_scaling_equality,
+            // ┌──────────┐
+            // │Fixed(==x)│
+            // └──────────┘
+            fixed_value_equality,
+            // ┌────────┐┌────────┐
+            // │Min(>=x)││Max(<=x)│
+            // └────────┘└────────┘
+            min_max_inequality,
+            // ┌───────────┐
+            // │Length(==x)│
+            // └───────────┘
+            length_equality,
+            // ┌───────────────┐
+            // │Percentage(==x)│
+            // └───────────────┘
+            percentage_equality,
+            // ┌────────────┐
+            // │Ratio(==x,y)│
+            // └────────────┘
+            ratio_equality,
+            // ┌────────┐┌────────┐
+            // │Min(==x)││Max(==x)│
+            // └────────┘└────────┘
+            min_max_equality,
+            // ┌─────────────────────┐
+            // │<= Proportional(x) =>│
+            // └─────────────────────┘
+            proportional_grower,
+            // ┌────────────┐
+            // │<= Min(x) =>│
+            // └────────────┘
+            grower,
+            // ┌       ┐
+            //  <= x =>
+            // └       ┘
+            space_grower,
+        ) = (
+            strength::create(1000.0, 1000.0, 999.0, 1.0), // REQUIRED - 1.0
+            strength::create(1000.0, 1000.0, 999.0, 1.0), // REQUIRED - 1.0
```

Comment:
Why not just use the comment version directly? They seem more readable / understanable?